axelrod/match_generator.py

```diff
@@ -207,3 +208,42 @@ def build_match_chunks(self):
 
     def __len__(self):
         return len(self.edges)
+
+
+class ProbEndSpatialMatches(SpatialMatches, ProbEndRoundRobinMatches):
+    """
+    A class that generates spatially-structured prob ending matches.
+    In these matches, players interact only with their neighbors rather than the
+    entire population. This reduces to a well-mixed population when the spatial
+    graph is a complete graph.
+
+    Parameters
+    ----------
+    players : list
+        A list of axelrod.Player objects
+    prob_end : float
+        The probability that a turn of a Match is the last
+    game : axelrod.Game
+        The game object used to score the match
+    repetitions : int
+        The number of repetitions of a given match
+    edges : list
+        A list of tuples containing the existing edges
+    """
+
+    def __init__(self, players, prob_end, game, repetitions, noise, edges):
+
```

Comment:
This looks like a repeat of the code for ```SpacialMatches.__init__``` If that's the case, then it ought to be a separate function that gets called by both init methods.